editor/components/block-drop-zone/index.js

```diff
@@ -53,10 +53,32 @@ function BlockDropZone( { index, isLocked, ...props } ) {
 		}
 	};
 
+	const onDrop = ( event, position ) => {
+		if ( ! event.dataTransfer ) {
+			return;
+		}
+
+		let uid, type, rootUID, fromIndex;
+
+		try {
+			( { uid, type, rootUID, fromIndex } = JSON.parse( event.dataTransfer.getData( 'text' ) ) );
+		} catch ( err ) {
+			return;
+		}
+
+		if ( type !== 'block' ) {
+			return;
+		}
+		const positionIndex = getInsertIndex( position );
+		const insertIndex = index && fromIndex < index && rootUID === props.rootUID ? positionIndex - 1 : positionIndex;
+		props.moveBlockToPosition( uid, rootUID, insertIndex );
+	};
+
 	return (
 		<DropZone
 			onFilesDrop={ onDropFiles }
 			onHTMLDrop={ onHTMLDrop }
+			onDrop={ onDrop }
```

Comment:
Unrelated: When does `onDrop` occur? On its name alone, I'd consider for _any_ drop, files or HTML included. But based on the separation of props, it seems more a fallback callback. The docs are very unclear (and by unclear, I mean non-existent).

Reply:
I think it's the former (called on every drop), but need to double check.